### PR TITLE
feat: update per API change for importing ruleset

### PR DESCRIPTION
**Pull request by formulahendry**

### CODE REVIEW

File: src/azure/ApiCenter/ApiCenterService.ts [hunk 1]
```diff
@@ -226,16 +226,40 @@ export class ApiCenterService {
...
     };
-    const response = await client.sendRequest(options);
-    return response;
+    let response = await client.sendRequest(options);
+
+    if (response.status === 202) {
+      const location = response.headers.get("Location");
+
+      if (!location) {
+        return { isSuccessful: false };
```
Comment (wenytang-ms) [inline src/azure/ApiCenter/ApiCenterService.ts:243]: do we need to log this as an error?

File: src/azure/ApiCenter/ApiCenterService.ts [hunk 2]
```diff
@@ -226,16 +226,40 @@ export class ApiCenterService {
...
+      options = {
+        method: "GET",
+        url: location,
+      };
+
+      const timeout = 30000; // 30 seconds in milliseconds
+      let startTime = Date.now();
+      do {
+        response = await client.sendRequest(options);
+        if (response.parsedBody?.status === "Succeeded") {
```
Comment (wenytang-ms) [inline src/azure/ApiCenter/ApiCenterService.ts:255]: it seems not a better condition, maybe the status should be defined as a constant or contract?

File: src/azure/ApiCenter/ApiCenterService.ts (hunk 1 above)
Comment (formulahendry) [reply src/azure/ApiCenter/ApiCenterService.ts:243]: Good point! Updated

File: src/azure/ApiCenter/ApiCenterService.ts (hunk 2 above)
Comment (formulahendry) [reply src/azure/ApiCenter/ApiCenterService.ts:255]: Currently, only 'status' is used, so not defined a contract

File: src/azure/ApiCenter/ApiCenterService.ts [hunk 3]
```diff
@@ -226,16 +226,40 @@ export class ApiCenterService {
...
+          return { isSuccessful: true };
```
Comment (wenytang-ms) [inline src/azure/ApiCenter/ApiCenterService.ts:256]: the response no need to return?

Comment (formulahendry) [reply src/azure/ApiCenter/ApiCenterService.ts:256]: Just return `ApiCenterRulesetImportResult`

File: src/azure/ApiCenter/ApiCenterService.ts [hunk 4]
```diff
@@ -226,16 +226,40 @@ export class ApiCenterService {
...
+      if (!location) {
+        return { isSuccessful: false };
+      }
+
+      options = {
+        method: "GET",
+        url: location,
+      };
+
+      const timeout = 30000; // 30 seconds in milliseconds
```
Comment (adashen) [inline src/azure/ApiCenter/ApiCenterService.ts:251]: Do we consider use retry times like 5 times? Or the service team recommend to use timeout?

Comment (formulahendry) [reply src/azure/ApiCenter/ApiCenterService.ts:251]: The 'importing spec' written by service team used this pattern, so I just follow it.

Comment (formulahendry) [reply src/azure/ApiCenter/ApiCenterService.ts]: Just synced with Alex yesterday, he recommand to set timeout as 60 seconds, as the job may be long.

**Review by wenytang-ms — APPROVED** [review]

**Pull request merged**